R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,288 @@
+             copy_rl = replace(value, province == "XZ" & fuel == "refined liquids", 0),
+             copy_g = replace(value, province == "XZ" & fuel == "gas", 0),
+             value = value + copy_rl + copy_g) %>%
+      select(-copy_g, -copy_rl) ->
+      L132.Mix_province_F_elec
+
+    # Aggregate by fuel compute each province's percentage, by fuel
+    L132.Mix_province_F_elec %>%
+      group_by(fuel, year) %>%
+      summarise(sum(value)) %>%
+      mutate(pct = value / sum(value),
+             pct = value / Sum,
+             fuel = replace(fuel, fuel == 'solar', 'solar PV')) %>%
+      ungroup %>%
+      select(-value, -Sum) %>%
+      replace_na(list(pct = 0)) ->
+      L123.pct_province_elec_F
+
+    # Electricity generation inputs by fuel and province
+    # Note that fuel inputs are only available for selected fuels; province_share_data uses this relevant subset
+    L123.in_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID & fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.in_EJ_province_elec_F
+    Fuel_in_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_in_Yh & year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.in_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      subset(select = c(-value, -pct, -GCAM_region_ID)) ->
+      L123.in_EJ_province_elec_F
+
+    # Electricity generation outputs by fuel and province
+    L123.out_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID & fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.out_EJ_province_elec_F
+    Fuel_out_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_out_Yh & year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.out_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      subset(select = c(-value, -pct, -GCAM_region_ID)) ->
+      L123.out_EJ_province_elec_F
+
+    # ELECTRICITY - OWNUSE
+    # Electricity own use by province
+    L126.in_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.in_EJ_CHINA_ownuse
+
+    L126.out_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.out_EJ_CHINA_ownuse
+
+    L123.in_EJ_CHINA_ownuse %>%
+      left_join_error_no_match(L123.out_EJ_CHINA_ownuse, by = 'year') %>%
+      mutate(value = value.x - value.y,
+             fuel = fuel.x,
+             sector = sector.x,
+             GCAM_region_ID = GCAM_region_ID.x) %>%
+      subset(select = c(GCAM_region_ID, sector, fuel, year, value)) ->
+      L123.net_EJ_CHINA_ownuse
+
+    # Then build table with each province's share of the national ownuse. Note that this is assumed invariant over time.
+    L123.out_EJ_province_elec_F %>%
+      group_by(province, year) %>%
+      summarise(sum(elec_F)) %>%
```

Comment:
`summarise(value = sum(elec_F)) %>%`
`ungroup %>%`
`group_by(year) %>%`
`mutate(value = value / sum(value) %>%`

Reply:
summarise can't be used before group_by(province, year)

Reply:
keep line 164, my comments start from line 165